savaslabs/savaslabs.github.io
# Issue #4673: create React app

Fixes issue [#4673](https://pm.savaslabs.com/issues/4673) and [#4675](https://pm.savaslabs.com/issues/4675)

## Summary of changes

1. Adds a skeleton React app and all the gulp tasks needed to bundle the JS and place it in the Jekyll `_site` directory.
2. Attaches the skeleton app to blog post pages to get us to a "hello world" state

## Notes

This is how I decided to structure the app:

![components](https://user-images.githubusercontent.com/8465998/31787286-43fcce2e-b4d9-11e7-84d5-ff7c8fd14d30.png)

There are two components that will be attached the DOM:

1. The App, which is a container component that holds:
    1. The Comments container component, which holds:
        1. Comment components
        2. The Comment Count component(so we can recreate the header "Comments (# of comments)"
    2. The Comment Form component, which may contain:
        1. A Message component (this might just go inside the App, not sure yet)
2. The Comment Count component, which needs to be attached to the top of the post page where we list the author, date, etc.

Please let me know if you have any feedback on this architecture, especially keeping in mind that we want to be able to easily implement the following future features:

- A text editor in the comment form
- Threaded comments

## To test

- Pull this code and run `npm install`
- Run `gulp clean` then `gulp serve`
- Check out a blog post page. At the top where the comment count used to be, you should see the text "Comment count goes here"
- Open up the comments expander at the bottom. You should see a bunch of hello world text coming from the app, the comment form, the comments container, and an individual comment
- Make sure you're not getting any console or gulp errors

Let me know what you think!


### Pull request reviewer

As the reviewer, I have verified the following:
